File: scala/src/main/scala/io/devorbit/sparklabs/batch/sales/OrdersPerCity.scala
package io.devorbit.sparklabs.batch.sales

import io.devorbit.sparklabs.batch.sales.Utils
import org.apache.spark.sql.functions._

object OrdersPerCity extends App {

  Utils.orders.join(Utils.customers, Seq("customer_id"), "inner")
    .groupBy(col("city"))
    .agg(count("*").as("total_orders_per_city"))
    .orderBy(desc("total_orders_per_city"))
    .show(false)

//  Thread.sleep(1000000)
}

File: python/sparklabs/batch/sales/dataset/orders_per_city.py
from pyspark.sql import functions as F

from sparklabs.batch.sales.dataset import utils


def run() -> None:
    (
        utils.orders.join(utils.customers, ["customer_id"], "inner")
        .groupBy(F.col("city"))
        .agg(F.count("*").alias("total_orders_per_city"))
        .orderBy(F.desc("total_orders_per_city"))
        .show(truncate=False)
    )


if __name__ == "__main__":
    run()
